File: Typescript/test/finalPositionOfSnake.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/finalPositionOfSnake';

test('finalPositionOfSnake Example 1', () => {
	let input0 = 2;
	let input1 = ["RIGHT", "DOWN"]
	let expected = 3;
	let output = solution.finalPositionOfSnake(input0, input1);
	expect(output).toBe(expected);
});

test('finalPositionOfSnake Example 2', () => {
	let input0 = 3;
	let input1 = ["DOWN", "RIGHT", "UP"]
	let expected = 1;
	let output = solution.finalPositionOfSnake(input0, input1);
	expect(output).toBe(expected);
});

test('finalPositionOfSnake Fail 1', () => {
	let input0 = 2;
	let input1 = ["DOWN", "RIGHT", "LEFT"]
	let expected = 2;
	let output = solution.finalPositionOfSnake(input0, input1);
	expect(output).toBe(expected);
});

File: Dart/test/finalPositionOfSnake_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/finalPositionOfSnake.dart';

void main() {
  test("test example 1", () {
    var input0 = 2;
    var input1 = ["RIGHT", "DOWN"];
    var expected = 3;

    final sln = Solution();
    var output = sln.finalPositionOfSnake(input0, input1);

    expect(output, expected);
  });

  test("test example 2", () {
    var input0 = 3;
    var input1 = ["DOWN","RIGHT","UP"];
    var expected = 1;

    final sln = Solution();
    var output = sln.finalPositionOfSnake(input0, input1);

    expect(output, expected);
  });
}
